### PR TITLE
Add SkyWalking Java Agent self observability dashboard

### PR DESCRIPTION
**Pull request by CzyerChen**
### https://github.com/apache/skywalking/issues/12595
- [x] Update the documentation to include this new feature.
- [x] If it's UI related, attach the screenshots below.

image
image

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes https://github.com/apache/skywalking/issues/12595.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java [hunk 1]
```diff
@@ -234,7 +234,13 @@ public enum Layer {
...
      * between application workloads such as application containers or processes.
      */
-    CILIUM_SERVICE(38, true);
+    CILIUM_SERVICE(38, true),
+
+    /**
+     * The self observability of SkyWalking Java Agent,
+     * which provides the abilities to measure the tracing performance and error statistics of plugins.
+     */
+    SO11Y_JAVA_AGENT(39, true);
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java:243]: I think we need a UI PR for this?

Comment (CzyerChen) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java:243]: OK

Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java:243]: How your preview works without UI change? Is that possible?

**Comment by wu-sheng** [comment]
I asked for a precision change for the agent side about duration cost by interceptor.
The current ms seems not precious enough, which would end all in 0.

My proposal is using nanosecond in agent and storage, but convert to ms or s as this is agent level metric.

**Comment by CzyerChen** [comment]
> I asked for a precision change for the agent side about duration cost by interceptor. The current ms seems not precious enough, which would end all in 0.
> 
> My proposal is using nanosecond in agent and storage, but convert to ms or s as this is agent level metric.

image

execution time storaged in `ns` and shown in `ms`.

**Comment by wu-sheng** [comment]
These are costs of all plugin executions in this minute, right? Could we add some tips to explain this metric and others?

Or do we have a well explained document?

**Comment by wu-sheng** [comment]
Please update the UI submodule. And I am going to merge https://github.com/apache/skywalking-java/pull/716, could you add the e2e accordingly?

**Comment by wu-sheng** [comment]
> Please update the UI submodule. And I am going to merge [apache/skywalking-java#716](https://github.com/apache/skywalking-java/pull/716), could you add the e2e accordingly?

Merged.

**Comment by CzyerChen** [comment]
> Please update the UI submodule. And I am going to merge [apache/skywalking-java#716](https://github.com/apache/skywalking-java/pull/716), could you add the e2e accordingly?

OK

**Comment by wu-sheng** [comment]
Suggestion, as we have Java agent in e2e, we could bump up java agent version, and add more checks to save resource costs of new e2e.

**Review by wankai123 — DISMISSED** [review]
LGTM

File: test/e2e-v2/cases/meter/docker-compose.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ version: '2.1'
 services:
   oap:
     environment:
-      SW_METER_ANALYZER_ACTIVE_FILES: spring-micrometer,batch-meter
+      SW_METER_ANALYZER_ACTIVE_FILES: spring-micrometer,batch-meter,java-agent
```
Comment (wu-sheng) [inline test/e2e-v2/cases/meter/docker-compose.yml:21]: Let's activate `java-agent` by default in the application.yml.

**Comment by wu-sheng** [comment]
There are some irrelevant test cases failing. Please recheck. 

Besides https://github.com/apache/skywalking/pull/12622#discussion_r1760615517, I think this is good.

**Comment by wu-sheng** [comment]
@CzyerChen @weixiang1862 This feature should be a good blog material. We could introduce what we add and why these are important.

**Comment by wu-sheng** [comment]
Could you share why Kafka needs to be disabled?

**Comment by wu-sheng** [comment]
Why some new metrics would affect that?

I just want to be certain, we are bot breaking anything.

**Comment by weixiang1862** [comment]
> Why some new metrics would affect that?
> 
> I just want to be certain, we are bot breaking anything.

Java agent so11y meters generated agent so11y service traffic which make the order of service list changes and generated new agent so11y traffic.

Original test cases has many service traffic actual and expected checks, it is big change to fix all service expected data, so disable `SW_METER_ACTIVE` by default and actived it in meter test case to validate so11y changes.

**Comment by wu-sheng** [comment]
So11y service is using the same service name and instance names, right?

I think we just build extra layer for each service?

**Comment by weixiang1862** [comment]
> So11y service is using the same service name and instance names, right?
> 
> I think we just build extra layer for each service?

Yes, the SO11Y_JAVA_AGENT service name and instance name is the same as GENERAL.

**Comment by weixiang1862** [comment]
> So11y service is using the same service name and instance names, right?
> 
> I think we just build extra layer for each service?

Metrics won't be effected, just service list values and order.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**